src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java

```diff
@@ -601,6 +609,18 @@ public void setYaml(String yaml) {
         this.yaml = yaml;
     }
 
+	public PodRetention getPodRetention() {
+		return podRetention;
+	}
+
+    @DataBoundSetter
+	public void setPodRetention(PodRetention podRetention) {
+        if (podRetention == null) {
+            podRetention = PodRetention.DEFAULT;
+        }
+		this.podRetention = podRetention;
+	}
```

Comment:
Retention on the pod template must always have a value, falling back to `DEFAULT`